osquery/utils/conversions/darwin/cfdictionary.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include "cfdictionary.h"
+
+namespace osquery {
+
+std::string getPropertiesFromDictionary(const CFDictionaryRef& dict,
```

Comment:
I'm fine either way, but does this make sense to take a pointer receiver and return a status? (I have no idea what our best practice is. But it's what it looks like `CFDictionaryGetValueIfPresent` uses. Though I guess that's CF, and not osquery.)

Either way, do we want tests?

Reply:
We want const references and not a pointer receiver. This also matches our current style. Also added tests, thanks!